### PR TITLE
Update data frame concept doc

### PR DESCRIPTION
**Pull request by marcusolsson**
**What this PR does / why we need it**:

Suggested improvements for data frame documentation.

### CODE REVIEW

**Review by bergquist — APPROVED** [review]
:+1: great work!

**Comment by kylebrandt** [comment]
Build failing, probably needs a merge from master.

**Review by kylebrandt — APPROVED** [review]

**Comment by kylebrandt** [comment]
@marcusolsson made some notes for "why" in https://gist.github.com/kylebrandt/4ba7b5f72738cc36c463e9dcc6e257b1 , could maybe incorporate into this PR if you think makes sense?

or can just do another after this is merged.

File: docs/sources/plugins/developing/dataframe.md [hunk 1]
```diff
@@ -145,7 +161,7 @@ This section contains links to technical reference and implementations of data f
 
 ### Apache Arrow
 
-The dataframe structure is inspired by and uses the [Apache Arrow Project](https://arrow.apache.org/). Javascript Data frames use Arrow Tables as the underlying structure, and the backend Go code serializes its Frames in Arrow Tables for transmission.
+The data frame structure is inspired by, and uses the [Apache Arrow Project](https://arrow.apache.org/). Javascript Data frames use Arrow Tables as the underlying structure, and the backend Go code serializes its Frames in Arrow Tables for transmission.
```
Comment (oddlittlebird) [inline docs/sources/plugins/developing/dataframe.md:164]:
```suggestion
The data frame structure is inspired by and uses the [Apache Arrow Project](https://arrow.apache.org/). Javascript Data frames use Arrow Tables as the underlying structure, and the backend Go code serializes its Frames in Arrow Tables for transmission.
```

**Comment by marcusolsson** [comment]
@kylebrandt Looks good! I'll integrate your notes in the next PR!

**Pull request merged**